### DISCUSSION TITLE
Migrate from 16.01 to 23.01-1

### DISCUSSION THREAD

**Discussion by PaweltK** [processone/ejabberd#4168] [Questions] [answered]
Hi,

I have  internal Jabber server on Ubuntu 16.04.7 LTS.
```
# ejabberdctl status
The node ejabberd@jabber is started with status: started
ejabberd 16.01 is running in that node
```

I want migrate to Debian 12
```
# ejabberdctl status
The node ejabberd@jabber is started with status: started
ejabberd 23.01-1 is running in that node
```

How to migrate mnesia database?
queue, archived messages are not important. Only accounts, passwords, contacts, etc.

Can you suggest what to do?
Of course, I tried backup/restore and dump/load but it doesn't work

**Comment by licaon-kter** [comment]
take it step by step: https://docs.ejabberd.im/admin/upgrade/#specific-version-upgrade-notes

**Comment by badlop** [comment, marked as answer]
Other update method you can try... it's very optimistic and may not work, but it only takes a few minutes to try:

1. First stop the old ejabberd,
2. Copy the mnesia spool directory to the new location (remove completely the new almost-empty database, don't mix files from old and new database).
3. Then start new ejabberd
4. Ideally, ejabberd will load your old mnesia database, notice it requires changes, and try to perform the required schema changes.

> Of course, I tried backup/restore and dump/load but it doesn't work

Did the new ejabberd complain that the mnesia database had a different erlang node name, or something like that? In that case see
https://docs.ejabberd.im/admin/guide/managing/#change-computer-hostname
